ouroboros-network-framework/src/Simulation/Network/Snocket.hs

```diff
@@ -689,7 +721,18 @@ mkSnocket state tr = Snocket { getLocalAddr
                                    (Just remoteAddress)
                                    (STBearerInfo bearerInfo))
             -- connection delay
-            threadDelay (biConnectionDelay bearerInfo)
+            unmask (threadDelay (biConnectionDelay bearerInfo `min` connectTimeout))
+              `catch` \(_ :: SomeException) ->
+                atomically $ modifyTVar (nsConnections state)
+                                        (Map.delete (normaliseId connId))
```

Comment:
Is this right thing to do? `SomeException` most likely mean an asynchronous exception in the application.  What we do here is automatic resource cleanup by on the OS side.  I am not sure if cancelling `connect` would behave like that on `*nix` platforms.

Let's ask Neil (@njd42) and @karknu.

Reply:
Yes, it means an asynchronous exception. Well I am not really sure if cancelling connect would behave like this but this is what I thought needs to be done to fix the test environment. Without this catch and cleanup thee test would fail with a `NotReleasedConnections` error from an `Half-Close`d peer.

Reply:
Why this  test failure is not a resource leak?

Reply:
I'm not sure I understand your question. The test was failing with a `NotReleasedConnections`. And it was due an `AsyncCancelled` exception thrown that would make everything die without cleaning up the state.

Reply:
`NotReleasedConnections` means that there is a resource leak in the test.  If throwing somewhere in the application an asynchronous exceptions leads to `NotReleasedConnections` this should rather be fixed in the application not the simulation environment.

Reply:
This could be as simple as using `bracket` somewhere at the top level (where the connection is created) or maybe it's an actual bug in the server.

Even if this is not a bug in connection-manager or server, such a fix would hide a true resource leak bug in diffusion (which should not leak any connections).

I bet this is an error somewhere inside the test itself (`prop_multinode_Sim`).  It doesn't seem very careful about resources.

Reply:
Okay I got confused, this patch appears here but shoudn't, this is probably due to some mistake of mine while rebasing. 

I fetched the original commit from you: https://github.com/input-output-hk/ouroboros-network/commit/d5026a1694bc6c3ea934a400293ae35c0a68fae2

Reply:
yes, that's my mistake.

Reply:
And the thing I was making confusion was this one: https://github.com/input-output-hk/ouroboros-network/pull/3309/commits/885b39aee1f1257ea6f016901a44ef677f4362bd#diff-9dc60e7be09dd9b4174122cea073cec55fb005ee452d64dafab6a12cfce6ed8bR728

Reply:
I take my word back.  We can catch an exceptions inside `connect` and update the simulation state. We're not closing the connection there (which we cannot do), we just remove it from the map of existing connections which is fine.